sevntu-checks/src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java

```diff
@@ -0,0 +1,1081 @@
+////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code for adherence to a set of rules.
+// Copyright (C) 2001-2019 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+////////////////////////////////////////////////////////////////////////////////
+
+package com.github.sevntu.checkstyle.checks.coding;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+import antlr.collections.AST;
```

Comment:
Use `DetailAST` over `AST`.

Reply:
I had wanted to do that already earlier. However that fires a CheckStyle violation:

```
[ERROR] src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java:[34] (imports) CustomImportOrder: Wrong lexicographical order for 'antlr.collections.AST' import. Should be before 'com.puppycrawl.tools.checkstyle.api.TokenTypes'.
[ERROR] src/main/java/com/github/sevntu/checkstyle/checks/coding/Jsr305AnnotationsCheck.java:[34] (imports) ImportOrder: Wrong order for 'antlr.collections.AST' import.
```

Reply:
Can you move import statement to location it demand ? Usually easy to fix

Reply:
Sorry, don't get this. Can you explain?

Reply:
@mbert I am asking for you to complete remove `antlr.collections.AST`. `DetailAST` is already in the import list, so you don't need to add anything there.
Current line 516 will have to change from `AST child` to `DetailAST child`.

Reply:
Ah, OK, of course! Done.